packages/shell-api/src/shell-instance-state.ts

```diff
@@ -483,17 +487,19 @@ export class ShellInstanceState {
       ): Promise<JSONSchema> => {
         const mongo = this.getMongoByConnectionId(connectionId);
         let docs: Document[] = [];
-        try {
-          docs = await mongo
-            ._getDb(databaseName)
-            .getCollection(collectionName)
-            ._getSampleDocsForCompletion();
-        } catch (err: any) {
-          if (
-            err?.code !== ShellApiErrors.NotConnected &&
-            err?.codeName !== 'Unauthorized'
-          ) {
-            throw err;
+        if (!disableSchemaSampling) {
```

Comment:
```suggestion
        if ((await this.evaluationListener.getConfig?.('disableSchemaSampling')) === false) {
```

This does require moving the key from `CliUserConfig` to `ShellUserConfig` but that's fine